integration/secret_cert_test.go

```diff
@@ -127,3 +129,130 @@ func TestCertificateUpload(t *testing.T) {
 
 	testenv.Test(t, f)
 }
+
+func TestCertificateUploadForOAS(t *testing.T) {
+	var (
+		tlsSecretName = "test-oas-cert-upload-secret"
+
+		testNS string
+		tykOas *v1alpha1.TykOasApiDefinition
+		tykEnv environment.Env
+		tykCtx context.Context
+	)
+	f := features.New("test certificate upload").
+		Setup(func(ctx context.Context, t *testing.T, envConf *envconf.Config) context.Context {
+			eval := is.New(t)
+			var err error
+
+			testNS, _ = ctx.Value(ctxNSKey).(string) // nolint: errcheck
+
+			// Obtain Environment configuration to be able to connect Tyk.
+			tykEnv, err = generateEnvConfig(ctx, envConf)
+			eval.NoErr(err)
+
+			res := semver.Compare(tykEnv.TykVersion, "v4.1")
```

Comment:
we'll we use 4.1 or 5.3 as the starting point for OAS. based on that we should align our code-base because we have discrepancy in oas versions now

Reply:
Makes sense